### PR TITLE
feat(coverage): add regular expression idempotency test

### LINKED ISSUE #5634: regular_expression: add idempotency test to `.to_string()`
For catching regressions such as https://github.com/oxc-project/oxc/pull/5631

### PR DESCRIPTION
**Pull request by Boshen**
closes #5634

### CODE REVIEW

File: tasks/coverage/src/driver.rs [hunk 1]
```diff
@@ -150,4 +156,41 @@ impl Driver {
...
+            let printed2 = match PatternParser::new(
+                &allocator,
+                &printed1,
+                ParserOptions {
+                    span_offset: 0,
+                    unicode_mode: flags.contains(RegExpFlags::U) || flags.contains(RegExpFlags::V),
+                    unicode_sets_mode: flags.contains(RegExpFlags::V),
+                },
+            )
+            .parse()
```
Comment (Boshen) [inline tasks/coverage/src/driver.rs:178]: These code is getting repeated a lot, we should export a top level function `parse_regular_expression(pattern, flags)` from the crate.

Comment (leaysgur) [reply tasks/coverage/src/driver.rs:178]: `oxc_regular_expression::Parser::new(&allocator, format!("/{printed1}/{flags}"), ParserOptions::default())` is still a bit verbose?

I will add it later. 👍🏻

**Review by leaysgur — APPROVED** [review]
Thanks for working this!

I will send 2 PRs later:

- Fix legacy octal printing
- Export `parse(pattern, flags)` API

File: tasks/coverage/src/driver.rs (hunk 1 above)
Comment (Boshen) [reply tasks/coverage/src/driver.rs:178]: This is an unwanted costly memory allocation ;-) format!("/{printed1}/{flags}")

**Comment by leaysgur** [comment]
> Export parse(pattern, flags) API

I've just remembered #5592 .

Once it get merged, we can construct options from `RegExpFlags` directly.

**Pull request merged**